jbpm-wb-case-mgmt/jbpm-wb-case-mgmt-backend/src/main/java/org/jbpm/workbench/cm/backend/server/RemoteCaseManagementServiceImpl.java

```diff
@@ -133,9 +134,39 @@ public void assignGroupToRole(final String serverTemplateId, final String contai
     }
 
     @Override
-    public void assignGroupAndUserToRole(final String serverTemplateId, final String containerId, final String caseId, final String roleName, final String user, final String group) {
-        client.assignGroupToRole(containerId, caseId, roleName, group);
-        client.assignUserToRole(containerId, caseId, roleName, user);
+    public void saveAssignment(final String serverTemplateId, final String containerId, final String caseId,
+                               final String roleName, final String users, final String groups) {
+        CaseRoleAssignmentSummary ras = getCaseInstance(serverTemplateId, containerId, caseId).getRoleAssignments()
+                .stream()
+                .filter(ra -> roleName.equals(ra.getName()))
+                .findFirst().orElse(null);
+        List<String> usersList = getListFromString(users);
+        List<String> groupsList = getListFromString(groups);
+        if (ras != null) {
+            if (ras.getUsers() != null && ras.getUsers().size() > 0) {
+                List<String> usersToRemove = new ArrayList();
+                usersToRemove.addAll(ras.getUsers());
+                usersToRemove.forEach(u -> removeUserFromRole(serverTemplateId, containerId, caseId, roleName, u));
+
+            }
+            if (ras.getGroups() != null && ras.getGroups().size() > 0) {
+                List<String> groupsToRemove = new ArrayList();
+                groupsToRemove.addAll(ras.getGroups());
+                groupsToRemove.forEach(g -> removeGroupFromRole(serverTemplateId, containerId, caseId, roleName, g));
+            }
+        }
+
+        usersList.forEach(u -> assignUserToRole(serverTemplateId, containerId, caseId, roleName, u));
+        groupsList.forEach(g -> assignGroupToRole(serverTemplateId, containerId, caseId, roleName, g));
+
+    }
+
+    private List<String> getListFromString(String stringList) {
+        if (stringList != null && stringList.trim().length() > 1) {
+            return Arrays.asList(stringList.split(",")).stream().map(s -> s.trim()).collect(toList());
+        } else {
+            return new ArrayList<String>();
```

Comment:
Unless you later need to add items to the list, `Collections.emptyList()` will work better in terms of both memory use and performance.